src/main/java/ch/njol/skript/bukkitutil/PaperEntityUtils.java

```diff
@@ -0,0 +1,171 @@
+			return;
+		}
+		for (LivingEntity entity : entities) {
+			if (!(entity instanceof Mob))
+				continue;
+			mobLookAt(target, headRotationSpeed, maxHeadPitch, (Mob) entity);
+		}
+	}
+
+	/**
+	 * Instruct a Mob (1.17+) or Players (1.19.1+) to look at a specific vector/location/entity.
+	 * Object can be a {@link org.bukkit.util.Vector}, {@link org.bukkit.Location} or {@link org.bukkit.entity.Entity}
+	 * THIS METHOD IS FOR 1.19.1+ ONLY. Use {@link lookAt(Object, Float, Float, LivingEntity...)} otherwise.
+	 * 
+	 * @param entityAnchor What part of the entity the player should face assuming the LivingEntity argument contains a player. Only for players.
+	 * @param target The vector/location/entity to make the livingentity or player look at.
+	 * @param headRotationSpeed The rotation speed at which the living entities will rotate their head to the target. Vanilla default values range from 10-50. Doesn't apply to players.
+	 * @param maxHeadPitch The maximum pitch at which the eyes/feet can go to. Doesn't apply to players.
+	 * @param entities The living entities to make look at something. Players can be involved in 1.19.1+
+	 */
+	public static void lookAt(LookAnchor entityAnchor, Object target, @Nullable Float headRotationSpeed, @Nullable Float maxHeadPitch, LivingEntity... entities) {
+		if (target == null || !LOOK_AT || !LOOK_ANCHORS)
+			return;
+		for (LivingEntity entity : entities) {
+			if (entity instanceof Player) {
+				Player player = (Player) entity;
+				if (target instanceof Vector) {
+					Vector vector = (Vector) target;
+					player.lookAt(vector.getX(), vector.getY(), vector.getZ(), LookAnchor.EYES);
+					player.lookAt(vector.getX(), vector.getY(), vector.getZ(), LookAnchor.FEET);
+				} else if (target instanceof Location) {
+					player.lookAt((Location) target, LookAnchor.EYES);
+					player.lookAt((Location) target, LookAnchor.FEET);
+				} else if (target instanceof Entity) {
+					player.lookAt((Entity) target, LookAnchor.EYES, entityAnchor);
+					player.lookAt((Entity) target, LookAnchor.FEET, entityAnchor);
+				}
+			} else if (entity instanceof Mob) {
+				mobLookAt(target, headRotationSpeed, maxHeadPitch, (Mob) entity);
+			}
+		}
+	}
+
+	public static class LookGoal implements Goal<Mob> {
+
+		private static final GoalKey<Mob> SKRIPT_LOOK_KEY = GoalKey.of(Mob.class, new NamespacedKey(Skript.getInstance(), "skript_entity_look"));
+		private static final EnumSet<GoalType> LOOK_GOAL = EnumSet.of(GoalType.LOOK);
+
+		private final float speed, maxPitch;
+		private final Object target;
+		private final Mob mob;
+		private int ticks = 0;
+
+		LookGoal(Object target, Mob mob, float speed, float maxPitch) {
+			this.maxPitch = maxPitch;
+			this.target = target;
+			this.speed = speed;
+			this.mob = mob;
+		}
+
+		@Override
+		public boolean shouldActivate() {
+			return ticks < 50;
+		}
+
+		@Override
+		public void tick() {
+			if (target instanceof Vector) {
```

Comment:
Would it be a good idea to save what type of entity it is in the constructor? I'm not sure how expensive it is to be calling instanceof this frequently. It might not matter though.